custom_components/aprilaire/humidifier.py

```diff
@@ -0,0 +1,225 @@
+        """Get current humidity."""
+        return self.coordinator.data.get(
+            Attribute.INDOOR_HUMIDITY_CONTROLLING_SENSOR_VALUE
+        )
+
+    async def async_set_humidity(self, humidity: int) -> None:
+        """Set the target dehumidification setpoint."""
+        await self.coordinator.client.set_dehumidification_setpoint(humidity)
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn on the dehumidifier."""
+        await self.async_set_humidity(self._last_target_humidity)
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn off the dehumidifier."""
+        await self.async_set_humidity(0)
+
+    async def async_set_mode(self, mode: str) -> None:
+        """Set the mode of the thermostat."""
+        await self.coordinator.client.read_scheduling()
+
+class AprilaireHumidifier(BaseAprilaireEntity, HumidifierEntity):
+    """Humidifier entity for Aprilaire."""
+
+    _attr_supported_features = HumidifierEntityFeature.MODES
+    _attr_device_class = HumidifierDeviceClass.HUMIDIFIER
+    _attr_min_humidity = 10
+    _attr_max_humidity = 50
+
+    def __init__(self, coordinator: AprilaireCoordinator) -> None:
+        self._last_target_humidity = None
+        self._last_is_on = None
+
+    @property
+    def action(self) -> HumidifierAction | None:
+        """Return the current action."""
+        
+        status = self.coordinator.data.get(
+            Attribute.HUMIDIFICATION_STATUS
+        )
+
+        if status == 0 or status == 1:
+            return HumidifierAction.IDLE
+        
+        if status == 2:
+            return HumidifierAction.HUMIDIFYING
+        
+        if status == 3:
+            return HumidifierAction.OFF
+        
+        raise ValueError(f"Unsupported humidification status {status}")
+
+    @property
+    def target_humidity(self) -> int:
+        """Return the target humidity."""
+        setpoint = self.coordinator.data.get(Attribute.HUMIDIFICATION_SETPOINT)
+        """Don't set last target humidity if off."""
+        self._last_is_on = setpoint != 0
+        self._last_target_humidity = setpoint if self._last_is_on else self._last_target_humidity
+        return self._last_target_humidity
+
+    @property
+    def is_on(self) -> bool:
+        """Return true if on."""
+        return self.target_humidity != 0
+
+    @property
+    def mode(self) -> str | None:
+        """Get humdification mode."""
+
+        available = self.coordinator.data.get(Attribute.HUMIDIFICATION_AVAILABLE)
+
+        if (available == 1):
+            return MODE_AUTO
+        
+        if (available == 2):
+            return MODE_MANUAL
+
+        raise ValueError(f"Unsupported humidification mode {available}")
+
+    @property
+    def available_modes(self) -> list[str] | None:
+        """Returns available modes."""
+
+        available = self.coordinator.data.get(Attribute.HUMIDIFICATION_AVAILABLE)
+
+        if (available == 1):
```

Comment:
@nberardi As far as I can tell, there is not a case where only auto OR manual is available. In all cases, it should be `[MODE_AUTO, MODE_MANUAL]`